modules/widget-visibility/widget-conditions.php

```diff
@@ -577,7 +591,7 @@ public static function filter_widget( $instance ) {
 									$condition_result = is_singular( substr( $rule['minor'], 10 ) );
 								} elseif ( $rule['minor'] == get_option( 'page_for_posts' ) ) {
 									// If $rule['minor'] is a page ID which is also the posts page
-									$condition_result = $wp_query->is_posts_page;
+									$condition_result = is_single();
```

Comment:
Same concern as above.  Breaking existing behavior by testing for something different.